hoot-test/src/main/cpp/hoot/test/main.cpp

```diff
@@ -599,42 +651,77 @@ void reportFailedTests(int failedTests, int totalTests)
   cout << failedTests << " of " << totalTests << " tests failed" << endl;
 }
 
-int main(int argc, char* argv[])
+void verifyValidationConfig()
 {
-  // Set the Qt hash seed to 0 for consistent test results.
-  conf().set(ConfigOptions().getHashSeedZeroKey(), true);
-  qSetGlobalQHashSeed(0);
+  // Verify that we have JOSM if running validation on test output is enabled.
+# ifndef HOOT_HAVE_JOSM
+  if (ConfigOptions().getTestValidationEnable())
+  {
+    throw TestConfigurationException("Test output validation requires compilation --with-josm.");
+  }
+# else
+  // Script tests require a dummy file written out so that they know its ok to run validation, since
+  // they can't do the compile time check.
+  const QString testValidationEnabledFile = "test-output/test-validation-enabled";
```

Comment:
Since config options are created and passed in per command in the script tests, this was the best I could come up with so that a script test knows whether it should try to validate output or not...write a dummy file if the conditions to validate are satisfied. If they aren't, delete the dummy file.